pkg/k2/constructs/template/execution/context.go

```diff
@@ -0,0 +1,131 @@
+package execution
+
+import (
+	"bytes"
+	"encoding"
+	"encoding/json"
+	"fmt"
+	"github.com/klothoplatform/klotho/pkg/templateutils"
+	"reflect"
+	"strconv"
+	"strings"
+	"text/template"
+)
+
+type (
+	// Context is an interface that defines the methods to execute a template and decode the result into a value
+	Context interface {
+		// ExecuteUnmarshal executes the template tmpl using data as input and unmarshals the value into v
+		ExecuteUnmarshal(tmpl string, data any, v any) error
+		// Unmarshal unmarshals the template result into a value
+		Unmarshal(data *bytes.Buffer, v any) error
+	}
+
+	DefaultExecutionContext struct{}
```

Comment:
Generally, interfaces are defined where they're used, not implemented. I also don't see any other implementations (maybe I'm missing it), so probably not necessary to preemptively make an interface for it.

Reply:
It's used in the `properties` package, which contains all the property implementations. I'm open to moving it if you feel strongly about it.

Reply:
I don't feel strongly, but the general practice of putting interfaces at the usage site makes them be way less bloated over time. It's a little weird compare to other languages like Java / TS, but it also makes it much easier to test since the surface area is smaller (the interface is what's used, not the union of all the uses).